socket.recv_into() types

I've noticed that the type in https://github.com/python/typeshed/blob/master/stdlib/2and3/socket.pyi#L555 :
```
    def recv_into(self, buffer: bytearray, nbytes: int,
        flags: int = ...) -> Any: ...
``` 
for `buffer` is `byterray`

while documentation specifies that `recv_into()` expects a [buffer object](https://docs.python.org/3/library/socket.html#socket.socket.recv_into), in particular `memoryview()` should work while it isn't a subtype of `bytearray()`.

Not sure if the correct solution would be to use `Union` here or if there is some abstract base class for `bytearray` and `memoryview`

Confirmed that this causes problems:
```
$ mypy --python-version 2.7 --custom-typeshed-dir . samples/recvintomv.py 
samples/recvintomv.py:6: error: Argument 1 to "recv_into" of "socket" has incompatible type "memoryview"; expected "bytearray"
$ cat samples/recvintomv.py 
import socket

sock = socket.socket()
mv = memoryview(b'hello')

sock.recv_into(mv, 1)
```

--- a/stdlib/2and3/socket.pyi
+++ b/stdlib/2and3/socket.pyi
@@ -8,6 +8,7 @@
 import sys
 from typing import Any, Iterable, Tuple, List, Optional, Union, overload, TypeVar
 
+_WriteBuffer = Union[bytearray, memoryview]
 
 # ----- variables and constants -----
 
@@ -550,9 +551,9 @@ class socket:
 
     # return type is an address
     def recvfrom(self, bufsize: int, flags: int = ...) -> Any: ...
-    def recvfrom_into(self, buffer: bytearray, nbytes: int,
+    def recvfrom_into(self, buffer: _WriteBuffer, nbytes: int,
                       flags: int = ...) -> Any: ...
-    def recv_into(self, buffer: bytearray, nbytes: int,
+    def recv_into(self, buffer: _WriteBuffer, nbytes: int,
                   flags: int = ...) -> Any: ...
     def send(self, data: bytes, flags: int = ...) -> int: ...
     def sendall(self, data: bytes, flags: int =...) -> None:
@@ -569,7 +570,7 @@ class socket:
     if sys.version_info >= (3, 3):
         def recvmsg(self, __bufsize: int, __ancbufsize: int = ...,
                     __flags: int = ...) -> Tuple[bytes, List[_CMSG], int, Any]: ...
-        def recvmsg_into(self, __buffers: Iterable[bytearray], __ancbufsize: int = ...,
+        def recvmsg_into(self, __buffers: Iterable[_WriteBuffer], __ancbufsize: int = ...,
                          __flags: int = ...) -> Tuple[int, List[_CMSG], int, Any]: ...
         def sendmsg(self, __buffers: Iterable[bytes], __ancdata: Iterable[_CMSG] = ...,
                     __flags: int = ..., __address: Any = ...) -> int: ...

